apollo-portal/src/test/java/com/ctrip/framework/apollo/portal/service/NamespaceBoAsyncServiceTest.java

```diff
@@ -0,0 +1,74 @@
+package com.ctrip.framework.apollo.portal.service;
+
+import com.ctrip.framework.apollo.common.dto.ItemDTO;
+import com.ctrip.framework.apollo.common.dto.ReleaseDTO;
+import com.ctrip.framework.apollo.portal.AbstractUnitTest;
+import com.ctrip.framework.apollo.portal.environment.Env;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author renjiahua
+ * @date 2021/1/17
+ */
+public class NamespaceBoAsyncServiceTest extends AbstractUnitTest {
+
+    @Mock
+    private ReleaseService releaseService;
+    @Mock
+    private ItemService itemService;
+    @InjectMocks
+    private NamespaceBoAsyncService namespaceBoAsyncService;
+
+    private String testAppId = "6666";
+    private String testClusterName = "default";
+    private String testNamespaceName = "application";
+    private Env testEnv = Env.DEV;
+
+    @Before
+    public void setup() {
+    }
+
+    @Test
+    public void testGetLatestReleaseAsync() throws ExecutionException, InterruptedException {
+        ReleaseDTO someRelease = new ReleaseDTO();
+        someRelease.setConfigurations("{\"a\":\"123\",\"b\":\"123\"}");
+        when(releaseService.loadLatestRelease(testAppId, testEnv, testClusterName, testNamespaceName))
+                .thenReturn(someRelease);
+        Future<Map<String, String>> latestReleaseAsync = namespaceBoAsyncService.getLatestReleaseAsync(testAppId, testEnv, testClusterName, testNamespaceName);
+        Map<String, String> asyncRelease = latestReleaseAsync.get();
+        assertEquals(2, asyncRelease.size());
+        assertEquals("123", asyncRelease.get("a"));
+        assertEquals("123", asyncRelease.get("b"));
+    }
+
+    @Test
+    public void testGetItemsAsynce() throws ExecutionException, InterruptedException {
```

Comment:
please add unit test to cover abnormal cases, e.g. when itemService.findItem throws exception, etc.

Reply:
Resolve

Reply:
Sorry but I don't see the unit test to cover abnormal cases, would you please take a look at this?